vector/src/main/java/im/vector/app/features/location/LocationSharingViewModel.kt

```diff
@@ -70,6 +75,22 @@ class LocationSharingViewModel @AssistedInject constructor(
         setUserItem()
         updatePin()
         compareTargetAndUserLocation()
+        checkPowerLevelsForLiveLocationSharing()
+    }
+
+    private fun checkPowerLevelsForLiveLocationSharing() {
+        PowerLevelsFlowFactory(room).createFlow()
+                .distinctUntilChanged()
+                .onEach {
+                    val powerLevelsHelper = PowerLevelsHelper(it)
+                    val canShareLiveLocation = EventType.STATE_ROOM_BEACON_INFO
+                            .map { beaconInfoType ->
+                                powerLevelsHelper.isUserAllowedToSend(session.myUserId, true, beaconInfoType)
+                            }
+                            .all { isUserAllowed -> isUserAllowed }
+
+                    setState { copy(canShareLiveLocation = canShareLiveLocation) }
+                }.launchIn(viewModelScope)
```

Comment:
Nit: You could use `setOnEach` for shorter code:
```kotlin
        PowerLevelsFlowFactory(room).createFlow()
                .distinctUntilChanged()
                .setOnEach {
                    val powerLevelsHelper = PowerLevelsHelper(it)
                    val canShareLiveLocation = EventType.STATE_ROOM_BEACON_INFO
                            .all { beaconInfoType ->
                                powerLevelsHelper.isUserAllowedToSend(session.myUserId, true, beaconInfoType)
                            }
                    copy(canShareLiveLocation = canShareLiveLocation)
                }
```

Reply:
I didn't know that. Perfect, done.